### PR TITLE
feat: [IOCOM-862,IOCOM-863] New DS on the message's attachments list

### PR DESCRIPTION
**Pull request by Vangaorth**
## Short description
This PR adds the attachment section to the message details, following the new DS styles and guidelines.
Simulator Screenshot - iPhone 15 - 2024-02-06 at 15 00 03

## List of changes proposed in this pull request
- New message details screen with attachment section
- Reference to the new ModuleAttachment (later to be replaced with the component in the DS package)
- Proper selectors for extracting relevant data
- Attachment download has been aligned with PN in order to download from the attachment list

## How to test
Using the io-dev-api-server, check that attachments are shown, downloaded and cancelled properly

### CODE REVIEW

File: ts/features/messages/store/reducers/thirdPartyById.ts [hunk 1]
```diff
@@ -84,10 +85,25 @@ export const messageMarkdownSelector = (
...
+) =>
+  pipe(
+    thirdPartyFromIdSelector(state, ioMessageId),
+    pot.toOption,
+    O.chainNullableK(
+      thirdPartyMessage => thirdPartyMessage.third_party_message.attachments
+    ),
+    O.getOrElse(
+      () => [] as ReadonlyArray<t.TypeOf<typeof ThirdPartyAttachment>>
+    )
```
Comment (adelloste) [inline ts/features/messages/store/reducers/thirdPartyById.ts:100]:
```suggestion
    O.getOrElse<ReadonlyArray<ThirdPartyAttachment>>(() => [])
```

File: ts/features/messages/components/MessageDetail/ModuleAttachment.tsx [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+  const IconOrActivityIndicatorComponent = () => {
+    if (isFetching) {
+      const activityIndicatorTestId = testID
+        ? `${testID}_activityIndicator`
+        : undefined;
+      return (
+        <ActivityIndicator
+          color={IOColors.blue}
+          testID={activityIndicatorTestId}
+        />
```
[the remark this answers is not in the document]
Comment (Vangaorth) [reply ts/features/messages/components/MessageDetail/ModuleAttachment.tsx]: Replaced

File: ts/features/messages/components/MessageDetail/ModuleAttachment.tsx [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+      );
+    }
+
+    return (
+      <Icon
+        name="chevronRightListItem"
+        color="blue"
+        size={IOListItemVisualParams.chevronSize}
+      />
```
[the remark this answers is not in the document]
Comment (Vangaorth) [reply ts/features/messages/components/MessageDetail/ModuleAttachment.tsx]: Replaced

File: ts/features/messages/store/reducers/thirdPartyById.ts (hunk 1 above)
Comment (Vangaorth) [reply ts/features/messages/store/reducers/thirdPartyById.ts]: Rewritten

File: ts/features/messages/__mocks__/attachment.ts
[the remark this answers is not in the document]
Comment (Vangaorth) [reply ts/features/messages/__mocks__/attachment.ts:1]: Replaced

**Review by adelloste — APPROVED** [review]

**Pull request merged**